tests/test_config.py

```diff
@@ -36,3 +38,13 @@ def test_specific_option():
     check(assert_type(pd.options.plotting.backend, str), str)
     # Just check assignment
     pd.options.plotting.backend = "matplotlib"
+
+
+def test_display_float_format():
+    check(
+        assert_type(pd.options.display.float_format, Optional[Callable[[float], str]]),
```

Comment:
It seems that python 3.8 doesn't like `Optional[Callable[[float], str]]` at runtime.  

But if you do `assert_type(pd.options.display.float_format, "Optional[Callable[[float], str]])"`, (i.e., surround the type in quotes), then it should pass the type checkers for all python versions as well as `pytest` .  You'd also have to remove the imports of `Callable` and `Optional`

Reply:
Hi Dr-Irv, thanks for the comment. Unfortunately if I specify the type as a string like this
```python
assert_type(pd.options.display.float_format, "Optional[Callable[[float], str]])"
```
and remove the imports, mypy produces some errors due to missing definitions (Python 3.10):
```
tests/test_config.py:43: error: Expression is of type "Optional[Callable[[float], str]]", not "Any"  [assert-type]
tests/test_config.py:43: error: Name "Optional" is not defined  [name-defined]
tests/test_config.py:43: note: Did you forget to import it from "typing"? (Suggestion: "from typing import Optional")
```
If I add back the imports then mypy succeeds, but then flake8 fails (because of the unused imports).

I think a solution could be to use the legacy `typing.Callable` class (that was replaced by PEP 585 in Python 3.9). This is deprecated, but should still work and can be changed back to `collections.abc.Callable` as soon as support for Python 3.8 is dropped (I would assume the legacy class will not be removed before then).

Reply:
I ran the CI with the current version, and it passed, so using `typing.Callable` does seem to be the way to go.